graph.py

```diff
@@ -0,0 +1,71 @@
+from itertools import product
+import random
+from collections import deque
+
+class Vertex():
+    def __init__(self,name:str,value=None,children:list=None):
+        if name is not None:
+            self.name = name
+        if value is not None:
+            self.value = value
+        else:
+            self.value = None
+
+        if children is not None:
+            self.children = children
+        else:
+            self.children = []
```

Comment:
I actually thought about this one (I had it as `set()` before and then changed it to `[]`). You can't do "random access" on a `set()` in the sense that we need: we need to uniformly draw `i` from `1` to `len(children)` and then get `children[i]`. I don't think you can do that on a `set` (outside of using a `dict` and then doing `.keys()` and then drawing from the resulting list). Further `list` is `O(1)` random access worst case because it's backed by something like [`std::vector`](http://stackoverflow.com/a/4637106). It's true that if we want this to be dynamic/online `[]` is not the right choice though.